service/history/service.go

```diff
@@ -42,7 +42,7 @@ type Service struct {
 	resource.Resource
 
 	status  int32
-	handler *Handler
+	handler *HandlerImpl
```

Comment:
Just use the interface rather than the implementation?

Reply:
At this point `handler` also use other methods (`Start`, `Stop`,...) defined on `HandlerImpl` but not on `Handler` interface. I wanted to keep this interface lean and only contain actual handler methods, not the utilities like `Start` and `Stop`.

Alternatively we could separate them into another interface and use that here instead. But for the time being and do not see immediate benefit for that and I wanted to keep this diff simple and mechanical.